### PR TITLE
Instant/design

### PR DESCRIPTION
**Pull request by peculiarity**

### CODE REVIEW

**Comment by krzkaczor** [comment]
Could you add dummy e2e test just making screenshots (so we can see them in vis reg) or add some storybooks?

**Comment by krzkaczor** [comment]
@peculiarity  ⬆️

File: src/instant/InstantView.tsx [hunk 1]
```diff
@@ -1,9 +1,141 @@
...
-    return <h1>Instant</h1>;
+    return (
+      <div className={styles.assetPicker}>
+        <Asset currency={this.props.asset} balance={this.props.balance}/>
+        <BigNumberInput
+          type="text"
+          className={styles.input}
+          mask={createNumberMask({
+            allowDecimal: true,
+            decimalLimit: 5,
```
Comment (kkostalkowicz) [inline src/instant/InstantView.tsx]: Should be parametrized (like `OfferMakeForm`)

File: src/instant/instant.ts [hunk 2]
```diff
@@ -321,30 +337,37 @@ function preValidate(state: InstantFormState): InstantFormState {
...
       field: receiveField,
       priority: 3,
+      placement: Position.TOP
     });
   }
 
   return {
     ...state,
-    messages,
+    messages: messages.sort(byPriority),
```
Comment (kkostalkowicz) [inline src/instant/instant.ts]: If sorting here, the relevant part of `offerMake` should be sorted as well.
Those two should have similar protocol.

File: src/AppContext.ts [hunk 3]
```diff
@@ -233,6 +212,30 @@ export function setupAppContext() {
   };
 }
 
+function instant(
+  orderbook$: Observable<Orderbook>,
+  balances$: Observable<Balances>
+) {
+  // TODO: find a way to wait for at least each  of the observables to emit a value
```
Comment (kkostalkowicz) [inline src/AppContext.ts]: Does the current implementation have a bug?

Comment (peculiarity) [reply src/AppContext.ts]: Good catch. That comment should be removed.

File: src/instant/instant.ts (hunk 2 above)
Comment (peculiarity) [reply src/instant/instant.ts]: But how come the offerMake should be the same. Those 2 parts are not connected. Also yesterday I've investigated the how the errors are handled in the `offerMake` and IMO it's too complex.

File: src/instant/InstantView.tsx (hunk 1 above)
Comment (peculiarity) [reply src/instant/InstantView.tsx]: Actually I've literally copy/pasted it from there so frankly I have no idea by parameterized in here. Please elaborate

File: src/instant/instant.ts (hunk 2 above)
Comment (peculiarity) [reply src/instant/instant.ts]: Actually in general we display 1 message only so I think we have to put only a single error message. I've chatted with MAciek and he is ok with that.

File: src/instant/InstantView.tsx (hunk 1 above)
Comment (kkostalkowicz) [reply src/instant/InstantView.tsx]: Tokens have different decimalLimits.
Hardcoded version works, but causes some input problems like ED-157

Comment (peculiarity) [reply src/instant/InstantView.tsx]: Yeah , just found it. Can we agree like this. I will put a TODO there and once I figure out how to build the list of all assets ( the token selector ) then I will parameterized it .. How does that sound to you ?

Comment (kkostalkowicz) [reply src/instant/InstantView.tsx]: Ok

Comment (peculiarity) [reply src/instant/InstantView.tsx]: I've pushed with the TODO in mind and removed the comment so can you recheck and approve :) Thanks

**Pull request merged**